internal/scanner/scanner.go

```diff
@@ -89,27 +89,24 @@ func (s *Scanner) ScanNamespace(nsName string) error {
 	// create PolicyReport
 	namespacedsReport := report.NewPolicyReport(namespace)
 	namespacedsReport.Summary.Skip = skippedNum
+	// old policy report to be used as cache
+	previousNamespacedReport, err := s.reportStore.GetPolicyReport(nsName)
+	if err != nil {
+		log.Error().Err(err).Str("namespace", nsName).
+			Msg("error getting PolicyReport from store")
+	}
 
 	// Iterate through all auditableResources. Each item contains a list of resources and the policies that would need
 	// to evaluate them.
 	for i := range auditableResources {
-		auditResource(&auditableResources[i], &s.resourcesFetcher, &s.httpClient, &namespacedsReport)
-		err = s.reportStore.AddPolicyReport(&namespacedsReport)
+		auditResource(&auditableResources[i], s.resourcesFetcher, &s.httpClient, &namespacedsReport, &previousNamespacedReport)
+		err = s.reportStore.SavePolicyReport(&namespacedsReport)
```

Comment:
question: do we want to remove the in memory store?

I thought that the idea was to have the in memory store so one can store each evaluation, and only do a client create or update when needing, blocking less.
I also thought we would keep the reportStore.ToJSON() for debugging and for consuming it by another component if needed.

In my opinion I would keep the in-mem store, and a new client store, with an interface for both.